### DISCUSSION TITLE
Ignore build step: turbo-ignore AND a restrict to branch

### DISCUSSION THREAD

**Discussion by Xhale1** [vercel/community#2597] [Help] [answered]
### Summary

The following two Vercel guides illustrate how to ignore builds from certain branches using a bash script, and how to ignore unnecessary builds in a turborepo monorepo using an npm script.

Is there a way to combine these two methods to only deploy changes on specified branches and still use `npx turbo-ignore` to only build if there are relevant changes?

- https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel
- https://vercel.com/changelog/intelligent-ignored-builds-using-turborepo

### Steps to Reproduce

Not relevant (I think)

**Comment by amyegan** [comment, marked as answer]
Hi @Xhale1! You can customize the Ignored Build Step script to handle this. For example: `if [ "$VERCEL_GIT_COMMIT_REF" == "main" ]; then npx turbo-ignore; else exit 0; fi` will ignore commits from branches other than `main` and use `turbo-ignore` to check if a commit to `main` should be deployed. In this example, you could change `exit 0` to `exit 1` if you want to allow all commits from other branches to build.
